torch/distributed/tensor/parallel/style.py

```diff
@@ -565,3 +569,40 @@ def __init__(
             if _prepare_output is not None
             else _get_prepare_output(output_layouts, use_local),
         )
+
+
+class PrepareModuleInput(ParallelStyle):
+    """
+    Only used to specify the input distribute spec for a module.
```

Comment:
we should make this documentation clearer, we should mention sth like:
PrepareModuleInput allows user to annotate the input torch.Tensor layouts to create DTensors base on input_layouts, and prepare the input DTensors be transformed to a desired layout base on output layouts. 

We should also explain each kwarg separately, and possibly add a small example demonstrate: annotate input torch.Tensor as a Sharded DTensor, then transform it to a Replicated DTensor

Reply:
Added comments, args doc and examples.